syntax = "proto3";
// Automatically generated by goyang https://github.com/openconfig/goyang
// compiled 2017-07-24T09:38:56Z
// do not delete the next line
// goyang-version 1
// module "device_config"
// namespace "urn:device-config"

// ff

option java_multiple_files = true;
option java_package = "mclab";
option java_outer_classname = "AP_proto";


package device_config;
service DeviceConfig {
  rpc EditConfig (EditConfigRequest) returns (EditConfigResponse);
  rpc GetConfig (GetConfigRequest) returns (GetConfigResponse);
  rpc Hello (HelloRequest) returns (HelloResponse);
  rpc Lock (LockRequest) returns (LockResponse);
}
message EditConfigRequest {
  string command = 1;
  string value = 2;
}
message EditConfigResponse {
  string channel = 1;
  string hw_mode = 2;
  string ip = 3;
  string password = 4;
  string power_on_off = 5;
  string ssid = 6;
}
message GetConfigRequest {
  string command = 1;
}
message GetConfigResponse {
  string channel = 1;
  string hw_mode = 2;
  string ip = 3;
  string password = 4;
  string power_on_off = 5;
  string ssid = 6;
}
message HelloRequest {
  string hello = 1;
}
message HelloResponse {
  string channel = 1;
  string hw_mode = 2;
  string ip = 3;
  string password = 4;
  string power_on_off = 5;
  string ssid = 6;
}
message LockRequest {
  bool locking = 1;
}
message LockResponse {
  bool result = 1;
}

message Devices {
  message AccessPoint {
    message Ap {
      string channel = 1;
      string hw_mode = 2;
      string ip = 3;
      string password = 4;
      string power_on_off = 5;
      string ssid = 6;
      uint32 uid = 7;
    }
    repeated Ap ap = 1;
  }
  AccessPoint access_point = 1;
  message LoraWanInformation {
    string temp = 1;
  }
  LoraWanInformation lora_wan_information = 2;
}

// Do not delete the lines below, they preserve tag information for goyang.
// goyang-tag Devices access_point/AccessPoint 1
// goyang-tag Devices lora_wan_information/LoraWanInformation 2
// goyang-tag Devices_AccessPoint ap/Ap[] 1
// goyang-tag Devices_AccessPoint_Ap channel/string 1
// goyang-tag Devices_AccessPoint_Ap hw_mode/string 2
// goyang-tag Devices_AccessPoint_Ap ip/string 3
// goyang-tag Devices_AccessPoint_Ap password/string 4
// goyang-tag Devices_AccessPoint_Ap power_on_off/string 5
// goyang-tag Devices_AccessPoint_Ap ssid/string 6
// goyang-tag Devices_AccessPoint_Ap uid/uint32 7
// goyang-tag Devices_LoraWanInformation temp/string 1
// goyang-tag EditConfigRequest command/string 1
// goyang-tag EditConfigRequest value/string 2
// goyang-tag EditConfigResponse channel/string 1
// goyang-tag EditConfigResponse hw_mode/string 2
// goyang-tag EditConfigResponse ip/string 3
// goyang-tag EditConfigResponse password/string 4
// goyang-tag EditConfigResponse power_on_off/string 5
// goyang-tag EditConfigResponse ssid/string 6
// goyang-tag GetConfigRequest command/string 1
// goyang-tag GetConfigResponse channel/string 1
// goyang-tag GetConfigResponse hw_mode/string 2
// goyang-tag GetConfigResponse ip/string 3
// goyang-tag GetConfigResponse password/string 4
// goyang-tag GetConfigResponse power_on_off/string 5
// goyang-tag GetConfigResponse ssid/string 6
// goyang-tag HelloRequest hello/string 1
// goyang-tag HelloResponse channel/string 1
// goyang-tag HelloResponse hw_mode/string 2
// goyang-tag HelloResponse ip/string 3
// goyang-tag HelloResponse password/string 4
// goyang-tag HelloResponse power_on_off/string 5
// goyang-tag HelloResponse ssid/string 6
// goyang-tag LockRequest locking/bool 1
// goyang-tag LockResponse result/bool 1
